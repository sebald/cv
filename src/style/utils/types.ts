/**
 * The values will be treated like the following:
 * - number -> "px" value
 * - string -> "rem" value
 */
export type FontValue = number | string;

export type ScaleConfig = {
  fontSize: FontValue;
  htmlFontSize: number;
  ratio: number;
};

export type LineHeightConfig = {
  fontSize: FontValue;
  htmlFontSize: number;
  baseline: number;
};

export type TypographyConfig = ScaleConfig & LineHeightConfig;

export type WithClassName = {
  className?: string;
};
