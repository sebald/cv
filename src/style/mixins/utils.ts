import { CSSProperties } from 'react';
import { StyledMixin } from '../theme';

export const propOrEmpty = <P extends keyof CSSProperties>(
  name: P
): StyledMixin<Pick<CSSProperties, P>> => (props: Pick<CSSProperties, P>) => {
  const p = props[name];
  return p ? { [name]: p } : {};
};
