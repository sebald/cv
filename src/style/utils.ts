import { theme } from './theme';

const { htmlFontSize } = theme;

export const px = (value: number) => `${value}px`;

export const pxToRem = (pixel: number, base = htmlFontSize) =>
  `${parseFloat((pixel / base).toFixed(5))}rem`;

export const remToPx = (rem: string, base = htmlFontSize) =>
  parseFloat(rem) * base;

export const ensurePx = (value: number | string, base = htmlFontSize) =>
  typeof value === 'string' ? remToPx(value, base) : value;
