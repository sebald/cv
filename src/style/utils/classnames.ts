export const mergeClassNames = (...names: (string | undefined)[]) => ({
  className: names
    .filter(x => typeof x === 'string')
    .join(' ')
    .trim(),
});
