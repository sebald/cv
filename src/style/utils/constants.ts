export const htmlFontSize = 16;

export const ratio = {
  perfectFourth: 1.333,
  perfectFifth: 1.5,
  goldenRatio: 1.618,
  majorSixth: 1.667,
};
