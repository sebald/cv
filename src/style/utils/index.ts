export * from './classnames';
export * from './constants';
export * from './createMixinComponent';
export * from './interpolation';
export * from './unit';
export * from './typography';
