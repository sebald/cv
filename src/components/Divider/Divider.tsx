import { color, suit, width } from '../../style';

export const Divider = suit<{ width?: number }>('hr')`
  border: 0;
  margin: 0 auto;

  height: 1px;
  ${width}

  background: ${color(palette => palette.grey['300'])};
  transform: translateY(-1px);
`;
