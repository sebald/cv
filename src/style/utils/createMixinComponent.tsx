import React from 'react';

import { styled } from '../styled-components';
import { StyledMixin } from '../theme';

import { mergeClassNames } from './classnames';
import { WithClassName } from './types';

/**
 * Idea stolen from:
 * https://github.com/jxnblk/grid-styled/tree/master/styled-space
 */
const StyledChildren: React.FC<WithClassName> = ({ className, children }) => {
  const styledChildren = React.Children.toArray(children).map((child) =>
    React.isValidElement<any>(child)
      ? React.cloneElement(
          child,
          mergeClassNames(child.props.className, className)
        )
      : child
  );

  return <>{styledChildren}</>;
};

export const createMixinComponent = <P extends {}>(
  ...mixins: StyledMixin<P>[]
) => styled(StyledChildren)([] as any, ...mixins);
