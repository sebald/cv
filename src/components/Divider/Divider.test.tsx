import * as React from 'react';

import { mountWithTheme } from '../../test.utils';
import { Divider } from './Divider';

test('render correctly', () => {
  const wrapper = mountWithTheme(<Divider />);
  expect(wrapper.find(Divider)).toMatchSnapshot();
});

test('render with custom width', () => {
  const wrapper = mountWithTheme(<Divider width={1 / 2} />);
  expect(wrapper.find(Divider)).toHaveStyleRule('width', '50%');
});
