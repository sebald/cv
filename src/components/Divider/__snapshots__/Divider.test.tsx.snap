// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
.c0 {
  border: 0;
  margin: 0 auto;
  height: 1px;
  background: #e0e0e0;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

<styled.hr>
  <hr
    className="c0"
  />
</styled.hr>
`;
